cmd/main.go

```diff
@@ -0,0 +1,53 @@
+package main
+
+import (
+    "fmt"
+    "net/http"
+    "os"
+    "os/signal"
+    "syscall"
+
+    bookSvc "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/service/books"
+    "github.com/King-Rullmann-Book-Club/books-api/pkg/v1/transport/books"
+
+    "github.com/go-kit/log"
+)
+
+func main() {
+    //	var (
+    //		httpAddr = flag.String("http.addr", ":8080", "HTTP listen address")
+    //	)
+    //	flag.Parse()
+
+    var logger log.Logger
+    {
+        logger = log.NewLogfmtLogger(os.Stderr)
+        logger = log.With(logger, "ts", log.DefaultTimestampUTC)
+        logger = log.With(logger, "caller", log.DefaultCaller)
+    }
```

Comment:
Could you explain this a bit? So far as I can tell, this is a scope wrapping some initalization functions, but it has no impact on the defintiion of logger. It is the same as 

```
var logger log.Logger
logger = log.NewLogfmtLogger(os.Stderr)
logger = log.With(logger, "ts", log.DefaultTimestampUTC)
logger = log.With(logger, "caller", log.DefaultCaller)
```

Is that right? Is this a stylistic choice or is there some effect I'm missing?

Reply:
I don't believe they are exactly the same. What happens inside the brace brackets, stays within the brace brackets :sunglasses: It's a scoping thing. It's common in Go to define a variable, in this case `logger` that is going to receive some manipulation, to define the variable using `var` and then modify the variable within a close scope. 

You'll see this with context as well:

```go
var ctx context.Context
{
  ctx = context.WithValue(context.Background(), "foo", "bar")
  ctx = context.WithValue(ctx, "hello", "world")
}
```